examples/firehose/src/main.rs

```diff
@@ -1,5 +1,6 @@
-use atrium_api::app::bsky::feed::post::Record;
```

Comment:
The firehose example update is in progress (#98), but the main purpose of this pull request should be to update the workspace, and I would appreciate it if you could leave this unchanged.

Reply:
This was the minimal change to keep Rust building, because the firehose was depending on an old published version of the crate, rather than the current state of the in-workspace crate. I can remove this change if you want, but then the example will be broken in `main` when this PR merges.

There's a separate question about whether you want the examples to be "what users should do with the latest published crates" or "what users should do with the crates as they exist in the branch the user is looking at". I personally find updating examples as `main` is changed useful, because it ensures that any library changes being made "work". But if instead you want these to be only relative to the published crates, then:
- that means I back out their usage of workspace dependencies,
- they probably shouldn't actually be part of the main library workspace, and
- there should be some process to ensure that these get updated after each new crate publish.

Reply:
Reverted these changes, by reverting the workspace inheritance conversion of `firehose`. That will need to be done in a subsequent PR.